crates/utils/src/settings/mod.rs

```diff
@@ -43,15 +45,25 @@ impl Settings {
   }
 
   pub fn get_database_url(&self) -> String {
-    let conf = &self.database;
-    format!(
-      "postgres://{}:{}@{}:{}/{}",
```

Comment:
A lot of the additions you added seem unecessary. Isn't this "postgres" part the only thing that needs to be added to the config? Then you could just add one optional config that defaults to `postgres://`

Reply:
No, the idea is to be able to supply the entire connection URI as one string because for example, as described in #2945, trying to use the unix domain socket via the configuration file is impossible under the status quo. See

https://github.com/LemmyNet/lemmy/blob/34c794e65e146c1490406f772e3d871d5ff42217/crates/utils/src/settings/structs.rs#L78

for how this has to be done, which should hopefully make it clear why I want to make this change.

Reply:
See also https://www.postgresql.org/docs/15/libpq-connect.html#id-1.7.3.8.3.6 for other things the current `format!()` call doesn't account for